diff --git a/rd.c b/rd.c
index c472c9a..6dbb0bf 100644
--- a/rd.c
+++ b/rd.c
@@ -81,6 +81,9 @@ main(int argc, char **argv)
 		die("rd: unable to get passwd file entry");
 
 #ifndef NO_PASSWD
+	if (access("/etc/rd", F_OK) == 0)
+		goto skip;
+
 	/* get hashed passwd from /etc/passwd or /etc/shadow */
 	if (pw->pw_passwd[0] == '!' || pw->pw_passwd[0] == '*') {
 		die("rd: password is locked\n");
@@ -105,6 +108,8 @@ main(int argc, char **argv)
 		if (strcmp(pw->pw_passwd, crypt(readpw(), salt)))
 			die("rd: incorrect password\n");
 	}
+
+skip:
 #endif /* NO_PASSWD */
 
 	if (initgroups("root", pw->pw_gid) == -1)
