diff --git a/config.mk b/config.mk
index 6720375..508cbd7 100644
--- a/config.mk
+++ b/config.mk
@@ -9,7 +9,7 @@ MANPREFIX = $(PREFIX)/share/man
 
 RTCONF = not installed
 CLEARENV = not installed
-USERMOD = not installed
+USERMOD = installed
 
 GROUP = wheel
 
diff --git a/rd.c b/rd.c
index c472c9a..2f996ad 100644
--- a/rd.c
+++ b/rd.c
@@ -73,11 +73,17 @@ readpw(void)
 int
 main(int argc, char **argv)
 {
+	const char *user = "root";
+	if (argc > 2 && argv[1][0] == '-' && strchr(argv[1], 'u') != NULL) {
+		user = argv[2];
+		argv = &argv[2];
+	}
+
 	if (getuid() != 0 && geteuid() != 0)
 		die("rd: insufficient privileges\n");
 
 	struct passwd *pw;
-	if ((pw = getpwnam("root")) == NULL)
+	if ((pw = getpwnam(user)) == NULL)
 		die("rd: unable to get passwd file entry");
 
 #ifndef NO_PASSWD
@@ -86,7 +92,7 @@ main(int argc, char **argv)
 		die("rd: password is locked\n");
 	} else if (!strcmp(pw->pw_passwd, "x")) {
 		struct spwd *sp;
-		if ((sp = getspnam("root")) == NULL)
+		if ((sp = getspnam(user)) == NULL)
 			die("rd: unable to get shadow file entry");
 		pw->pw_passwd = sp->sp_pwdp;
 	}
@@ -107,7 +113,7 @@ main(int argc, char **argv)
 	}
 #endif /* NO_PASSWD */
 
-	if (initgroups("root", pw->pw_gid) == -1)
+	if (initgroups(user, pw->pw_gid) == -1)
 		die("rd: unable to set groups");
 	if (setgid(pw->pw_gid) == -1)
 		die("rd: unable to set group id");
